repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -609,11 +609,31 @@ private void addLimitlessVerticesToResult(boolean isInput, int depth, Set<String
         }
     }
 
-    private void processLastLevel(AtlasVertex currentVertex, boolean isInput, AtlasLineageInfo ret) {
+    private void processLastLevel(AtlasVertex currentVertex, boolean isInput, AtlasLineageInfo ret, AtlasLineageContext lineageContext) {
         List<AtlasEdge> processEdges = vertexEdgeCache.getEdges(currentVertex, IN, isInput ? PROCESS_OUTPUTS_EDGE : PROCESS_INPUTS_EDGE);
+        processEdges = lineageContext.getIgnoredProcesses() != null ? eliminateIgnoredProcesses(processEdges, isInput, lineageContext) : processEdges;
```

Comment:
lineageContext.getIgnoredProcesses() != null --> CollectionUtils.isNotEmpty(lineageContext.getIgnoredProcesses())
checks for empty collections as well

Reply:
addressed